cloudinit/net/network_state.py

```diff
@@ -683,6 +692,24 @@ def handle_ethernets(self, command):
                     str(cfg),
                 )
             phy_cmd["mac_address"] = mac_address
+
+            # Determine the name of the interface by using one of the
+            # following in the order they are listed:
+            #   * set-name
+            #   * interface name looked up by mac
+            #   * value of "eth" key from this loop
+            name = eth
+            set_name = cfg.get("set-name", None)
+            if set_name:
+                name = set_name
+            elif mac_address and ifaces_by_mac:
+                lcase_mac_address = mac_address.lower()
```

Comment:
Nit: this line could be removed and `.lower()` could be called in the if statement.

Reply:
Yes, good point. Ignore my comment.